win32/src/PyWinTypesmodule.cpp

```diff
@@ -674,57 +677,70 @@ PyObject *PyWinObject_FromRECT(LPRECT prect)
     return Py_BuildValue("llll", prect->left, prect->top, prect->right, prect->bottom);
 }
 
-// Buffer conversion functions that use DWORD for length
-BOOL PyWinObject_AsReadBuffer(PyObject *ob, void **buf, DWORD *buf_len, BOOL bNoneOk)
+// replacement for PyWinObject_AsReadBuffer and PyWinObject_AsWriteBuffer
+BufferView::BufferView()
 {
+    memset(&m_view, 0, sizeof(m_view));
+}
+
+bool BufferView::init(PyObject *ob, bool bWrite, bool bNoneOk)
+{
+    release();
+    memset(&m_view, 0, sizeof(m_view));
     if (ob == Py_None) {
         if (bNoneOk) {
-            *buf_len = 0;
-            *buf = NULL;
-            return TRUE;
+            // using Py_None as sentinel, leaving m_view's buf and len equal to 0
+            m_view.obj = Py_None;
         }
-        PyErr_SetString(PyExc_TypeError, "Buffer cannot be None");
-        return FALSE;
+        else
```

Comment:
@mhammond  I am not sure whether I understood. Do you mean you would prefere
```
if (condition) {
    do_something();
    } else
```
over
```
if (condition) {
    do_something();
    }
    else
```
?
If that is what you want, I can easily do so, but it would baffle me since the rest of the file does it differently.

Reply:
Hopefully solved by updated PR